src/main/java/by/exadel/internship/service/FormService.java

```diff
@@ -19,10 +27,51 @@
     private final FormMapper mapper;
     private final FormRepository formRepository;
 
-    public void saveForm(FormRegisterDTO formRegisterDTO) {
+    @Value("${file.path}")
+    private String filePath;
+
+    public Form process(FormRegisterDTO form, MultipartFile file) {
+        MDC.put("className", FormService.class.getSimpleName());
+        log.info("Try to save form");
+        if (file != null) {
```

Comment:
Simplify this part. Move saving form as a first step, which will be general for both cases (with or without file). And inside if leave only part related upload file.

Reply:
done